File: CRUD/PHP/Ajax V3/entities/Producto.php
<?php

class Producto
{
    
    private $id;
    private $nombre;
    private $direccion;
    private $telefono;
    private $id_proveedor;
    private $fecha_registro;

    private $proveedor;
    
    public function __construct()
    {
        $this->id     = "";
        $this->nombre = "";
        $this->descripcion     = "";
        $this->precio          = "";
        $this->fecha_registro  = "";
        $this->id_proveedor    = "";

        $this->proveedor = "";
    }
    
    public static function CreateProducto($id, $nombre, $descripcion, $precio, $id_proveedor, $fecha_registro)
    {
        $instance                  = new self();
        $instance->id     = $id;
        $instance->nombre = $nombre;
        $instance->descripcion     = $descripcion;
        $instance->precio          = $precio;
        $instance->id_proveedor    = $id_proveedor;
        $instance->fecha_registro  = $fecha_registro;
        return $instance;
    }
    
    public function getId()
    {
        return $this->id;
    }
    
    public function setId($id)
    {
        $this->id = $id;
    }
    
    public function getNombre()
    {
        return $this->nombre;
    }
    
    public function setNombre($nombre)
    {
        $this->nombre = $nombre;
    }
    
    public function getDescripcion()
    {
        return $this->descripcion;
    }
    
    public function setDescripcion($descripcion)
    {
        $this->descripcion = $descripcion;
    }
    
    public function getPrecio()
    {
        return $this->precio;
    }
    
    public function setPrecio($precio)
    {
        $this->precio = $precio;
    }
    
    public function getFecha_registro()
    {
        return $this->fecha_registro;
    }
    
    public function setFecha_registro($fecha_registro)
    {
        $this->fecha_registro = $fecha_registro;
    }
    
    public function getId_proveedor()
    {
        return $this->id_proveedor;
    }
    
    public function setId_proveedor($id_proveedor)
    {
        $this->id_proveedor = $id_proveedor;
    }
    
    public function getProveedor()
    {
        return $this->proveedor;
    }
    
    public function setProveedor($proveedor)
    {
        $this->proveedor = $proveedor;
    }

    public function __destruct()
    {
        $this->id     = "";
        $this->nombre = "";
        $this->descripcion     = "";
        $this->precio          = "";
        $this->fecha_registro  = "";
        $this->id_proveedor    = "";

        $this->proveedor = "";
    }
    
    public function toString()
    {
        return "Producto{" . "id=" . $this->id . ", nombre=" . $this->nombre . ", descripcion=" . $this->descripcion . ", precio=" . $this->precio . ", fecha_registro=" . $this->fecha_registro . ", id_proveedor=" . $this->id_proveedor . '}';
    }
    
}

?>

File: CRUD/Java usando NetBeans/Aplicacion Web usando MVC Spring V2/SistemaWebSpring/src/main/java/sistema/models/Producto.java
// Written by Ismael Heredia in the year 2017
package sistema.models;

import javax.validation.constraints.NotNull;
import org.hibernate.validator.constraints.NotEmpty;

public class Producto {

    int id;

    @NotEmpty
    String nombre;

    @NotEmpty
    String descripcion;

    @NotNull
    double precio;

    @NotNull
    int id_proveedor;

    String fecha_registro;

    public Proveedor proveedor;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public int getId_proveedor() {
        return id_proveedor;
    }

    public void setId_proveedor(int id_proveedor) {
        this.id_proveedor = id_proveedor;
    }

    public String getFecha_registro() {
        return fecha_registro;
    }

    public void setFecha_registro(String fecha_registro) {
        this.fecha_registro = fecha_registro;
    }

    public Proveedor getProveedor() {
        return proveedor;
    }

    public void setProveedor(Proveedor proveedor) {
        this.proveedor = proveedor;
    }

    public Producto() {

    }

    public Producto(int id, String nombre, String descripcion, double precio, int id_proveedor, String fecha_registro) {
        this.id = id;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.id_proveedor = id_proveedor;
        this.fecha_registro = fecha_registro;
    }

    @Override
    public String toString() {
        return "Producto{" + "id=" + id + ", nombre=" + nombre + ", descripcion=" + descripcion + ", precio=" + precio + ", id_proveedor=" + id_proveedor + ", fecha_registro=" + fecha_registro + ", proveedor=" + proveedor + '}';
    }

}
